File: java/demo/src/main/java/cn/huidu/device/demo/demo1_deviceInfod/RebootDevice.java
package cn.huidu.device.demo.demo1_deviceInfod;

import cn.huidu.device.sdk.Device;

public class RebootDevice {
    public static void main(String[] args) {
        // Replace with your own parameters
        // String host = "${YourHost}";
        // String sdkKey = "${YourSdkKey}";
        // String sdkSecret = "${YourSdkSecret}";
        // Config.InitSdk(host, sdkKey, sdkSecret);

        // 1. Instantiate device interface object
        Device device = new Device();
        // 2. Call interface to reboot device
        String jsonResult = device.device(new String[] { "C16-D21-015BD" }, "rebootDevice");
        // 3. Print result
        System.out.println(jsonResult);

    }

}

File: python/demos/demo1_deviceInfo/RebootDevice.py
from sdk.common.Config import Config
from sdk.Device import Device
import json

def main():
    # with your own parameters
    # host = "${YourHost}"
    # sdk_key = "${YourSdkKey}"
    # sdk_secret = "${YourSdkSecret}"
    # Config.init_sdk(host, sdk_key, sdk_secret)

    host = "http://192.168.90.153:30080"
    sdk_key = "a7fa6795aaa891e2"
    sdk_secret = "696b35024ea5c2ba0db2876663254a78"
    Config.init_sdk(host, sdk_key, sdk_secret)

    # Instantiate a device interface object
    device = Device()

    # Device reboot
    device_id = ["C16L-D00-A000F"]
    result = device.device(device_id, "rebootDevice")

    # Print the result
    print("API Respond to the results:")
    print(json.dumps(result, indent=2))

if __name__ == "__main__":
    main()
